### DISCUSSION TITLE
ChipsInput

### DISCUSSION THREAD

**Discussion by cbhouston** [anvilistas/anvil-extras#227] [Ideas]
Would love the ChipsInput to take an actual Chip component, can change the background color of the chip going in. Would love to show the colors showing where a chip is within a workflow process. Would this be possible now? Using ChipsInput ?

**Comment by s-cork** [comment]
Do you have a proposal for the api?

One possible solution is to expose more chip attributes like:
- `foreground`
- `background`
Along with extra attributes
- `highlight_foreground`
- `highlight_background`

Another would be to expose an event
- `chip_highlight` where one of the paramaters is the chip itself that is being highlighted
This might be combined with a `chip_template` where you can provide a styled chip at runtime.

```python
    self.chips_input.chip_template = Chip(foreground='white', background='theme:Secondary 500')

    def chips_input_chip_highlight(self, chip, **event_args):
        chip.foreground = 'blue'
        chip.background = 'white'
```

**Comment by cbhouston** [comment]
I wish I had the python skills to recommend a direction, I do not as only been in it this year.  I would think attributes would be the best way to go. Or ChipsInput accepts the Chip that already has the attributes.

**Comment by s-cork** [comment]
Apart from colors is there anything else that is missing?

**Comment by cbhouston** [comment]
Colors came out something that would be very handy for quick color status of the various chips

But..... if you wanna pack a lot of punch to a chip, the pop-over ability would be awesome too. highlight the chip and it pops up extra options, or extra information, or anything like that.  I am sure that is much much harder to implement just by judging the doc online.
